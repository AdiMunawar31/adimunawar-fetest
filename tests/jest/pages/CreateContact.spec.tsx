import React from "react";
import renderer from "react-test-renderer";
import CreateContact from "@pages/Dashboard/Create";
import { MemoryRouter } from "react-router-dom";

jest.mock("react-redux", () => ({
  useDispatch: () => jest.fn(),
  useNavigate: () => jest.fn(),
  useSelector: () => [
    {
      id: "1",
      firstName: "John",
      lastName: "Doe",
      photo: "john.png",
    },
    {
      id: "2",
      name: "Jane",
      lastName: "Smith",
      photo: "janesmith.png",
    },
  ],
}));

describe("CreateContact component", () => {
  it("renders the CreateContact component", () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={["/create-contact"]}>
        <CreateContact />
      </MemoryRouter>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
