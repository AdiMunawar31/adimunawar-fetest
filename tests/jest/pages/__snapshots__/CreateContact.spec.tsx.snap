// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateContact component renders the CreateContact component 1`] = `
<div
  className="bg-gradient-to-br from-cyan-300 via-cyan-100 to-blue-900 w-screen h-screen sm:p-5"
>
  <div
    className="bg-white border border-gray-200 rounded flex h-full"
  >
    <div
      className="w-full sm:w-1/2 md:w-1/3 lg:w-1/4 h-full overflow-y-auto scrollbar-thumb-blue scrollbar-thumb-rounded scrollbar-track-blue-lighter scrollbar-w-2 scrolling-touch"
    >
      <div
        className="border-b border-gray-200 p-3 relative flex justify-between"
      >
        <button
          className="flex items-start select-none font-semibold focus:outline-none"
        >
          <a
            className="flex"
            href="/dashboard"
            onClick={[Function]}
          >
            <img
              alt="logo"
              src="/contact.png"
              width={40}
            />
            <span
              className="text-lg text-cyan-400 font-bold ml-2 mt-2"
            >
              DiContact
            </span>
          </a>
        </button>
        <a
          className="bg-red-200 p-2 flex items-center rounded-md"
          href="/"
          onClick={[Function]}
          title="Shutdown"
        >
          <svg
            className="text-red-500 text-xl font-bold"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6.26489 3.80698L7.41191 5.44558C5.34875 6.89247 4 9.28873 4 12C4 16.4183 7.58172 20 12 20C16.4183 20 20 16.4183 20 12C20 9.28873 18.6512 6.89247 16.5881 5.44558L17.7351 3.80698C20.3141 5.61559 22 8.61091 22 12C22 17.5228 17.5228 22 12 22C6.47715 22 2 17.5228 2 12C2 8.61091 3.68594 5.61559 6.26489 3.80698ZM11 12V2H13V12H11Z"
            />
          </svg>
        </a>
      </div>
      <div
        className="flex items-center justify-between text-sm border-b border-gray-200"
      >
        <div>
          <button
            className="py-3 w-20 uppercase font-semibold select-none h-full focus:outline-none border-b border-black"
          >
            Primary
          </button>
          <button
            className="py-3 w-20 uppercase text-gray-400 select-none h-full focus:outline-none border-b border-transparent"
          >
            General
          </button>
        </div>
        <button
          className="p-3 font-semibold text-blue-500 select-none h-full focus:outline-none border-b border-transparent"
        >
          2
           Contacts
        </button>
      </div>
      <ul
        className="py-1 contact-list"
      >
        <li
          className="border-b cursor-pointer contact-item"
        >
          <div
            className="flex items-center justify-between w-full px-4 py-0.5 select-none hover:bg-gray-100 focus:outline-none relative"
          >
            <div
              className="flex pr-20"
              onClick={[Function]}
            >
              <img
                alt="Photo profile"
                className="w-12 mr-3 h-12 border-cyan-500 rounded-full border-2 p-0.5"
                src="janesmith.png"
              />
              <div
                className="block"
              >
                <h3
                  className="font-semibold capitalize -mb-1 truncate w-40"
                >
                   
                  Smith
                </h3>
                <span
                  className="text-xs text-gray-500"
                >
                  Age: 
                </span>
              </div>
            </div>
            <button
              className="p-2 top-1/2 transform mt-8 -translate-y-1/2"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
                />
              </svg>
            </button>
          </div>
        </li>
        <li
          className="border-b cursor-pointer contact-item"
        >
          <div
            className="flex items-center justify-between w-full px-4 py-0.5 select-none hover:bg-gray-100 focus:outline-none relative"
          >
            <div
              className="flex pr-20"
              onClick={[Function]}
            >
              <img
                alt="Photo profile"
                className="w-12 mr-3 h-12 border-cyan-500 rounded-full border-2 p-0.5"
                src="john.png"
              />
              <div
                className="block"
              >
                <h3
                  className="font-semibold capitalize -mb-1 truncate w-40"
                >
                  John
                   
                  Doe
                </h3>
                <span
                  className="text-xs text-gray-500"
                >
                  Age: 
                </span>
              </div>
            </div>
            <button
              className="p-2 top-1/2 transform mt-8 -translate-y-1/2"
              onClick={[Function]}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"
                />
              </svg>
            </button>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="hidden sm:w-1/2 md:w-2/3 lg:w-3/4 border-l border-gray-200 sm:flex items-center justify-center text-center overflow-y-auto scrollbar-thumb-blue scrollbar-thumb-rounded scrollbar-track-blue-lighter scrollbar-w-2 scrolling-touch"
    >
      <div
        className="container mx-auto scrollbar-thumb-blue scrollbar-thumb-rounded scrollbar-track-blue-lighter scrollbar-w-2 scrolling-touch"
      >
        <div
          className="w-full bg-white p-5 rounded-md"
        >
          <div
            className="flex mt-8 text-lg mx-6 text-start p-5 bg-blue-50 font-semibold text-cyan-500"
          >
            <svg
              className="mt-1 mr-3"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 0H4v2h16V0zM4 24h16v-2H4v2zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-8 2.75c1.24 0 2.25 1.01 2.25 2.25s-1.01 2.25-2.25 2.25S9.75 10.24 9.75 9 10.76 6.75 12 6.75zM17 17H7v-1.5c0-1.67 3.33-2.5 5-2.5s5 .83 5 2.5V17z"
              />
            </svg>
            <span>
              Add Contact Data Form
            </span>
          </div>
          <div
            className="m-7"
          >
            <form
              onSubmit={[Function]}
            >
              <div
                className="mb-3"
              >
                <label
                  className="block mb-2 text-sm text-gray-600 text-start"
                  htmlFor="firstName"
                >
                  First Name
                   :
                </label>
                <input
                  className="w-full p-3 placeholder-gray-400 border border-gray-400 rounded-md focus:outline-none focus:ring focus:ring-blue-100 focus:border-blue-400 "
                  name="firstName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="e.g John"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="mb-3"
              >
                <label
                  className="block mb-2 text-sm text-gray-600 text-start"
                  htmlFor="lastName"
                >
                  Last Name
                   :
                </label>
                <input
                  className="w-full p-3 placeholder-gray-400 border border-gray-400 rounded-md focus:outline-none focus:ring focus:ring-blue-100 focus:border-blue-400 "
                  name="lastName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="e.g Doe"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="mb-3"
              >
                <label
                  className="block mb-2 text-sm text-gray-600 text-start"
                  htmlFor="age"
                >
                  Age
                   :
                </label>
                <input
                  className="w-full p-3 placeholder-gray-400 border border-gray-400 rounded-md focus:outline-none focus:ring focus:ring-blue-100 focus:border-blue-400 "
                  name="age"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="e.g 31"
                  type="number"
                  value=""
                />
              </div>
              <div
                className="mb-3"
              >
                <label
                  className="block mb-2 text-sm text-gray-600 text-start"
                  htmlFor="photo"
                >
                  Photo
                   :
                </label>
                <input
                  className="w-full p-3 placeholder-gray-400 border border-gray-400 rounded-md focus:outline-none focus:ring focus:ring-blue-100 focus:border-blue-400 "
                  name="photo"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="e.g https://example-photos.png"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="w-4/12 gap-2 my-8 flex"
              >
                <button
                  className="w-full py-2 px-3 text-white rounded-md focus:bg-blue-600 focus:outline-none bg-gray-400 cursor-not-allowed"
                  disabled={true}
                  type="submit"
                >
                  Add Contact
                </button>
                <a
                  className="w-full py-2 px-3 text-white rounded-md focus:bg-red-600 focus:outline-none bg-red-500 hover:bg-red-400"
                  href="/dashboard"
                  onClick={[Function]}
                >
                  Cancel
                </a>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
